cmd/kube-proxy/app/server.go

```diff
@@ -312,19 +316,19 @@ func (o *Options) Run() error {
 	}
 
 	o.proxyServer = proxyServer
-	return o.runLoop()
+	return o.runLoop(ctx)
 }
 
 // runLoop will watch on the update change of the proxy server's configuration file.
 // Return an error when updated
-func (o *Options) runLoop() error {
+func (o *Options) runLoop(ctx context.Context) error {
 	if o.watcher != nil {
 		o.watcher.Run()
 	}
 
 	// run the proxy in goroutine
 	go func() {
-		err := o.proxyServer.Run()
+		err := o.proxyServer.Run(ctx)
```

Comment:
I'm from mobile, but iirc this `errCh` is important here for context cancellation

Reply:
Yes, currently, when kube-proxy configuration changed, healthz server failed or metrics server failed, we'll both sent events to errCh, then `runLoop` will return with error, and proxy will exit(1), but we didn't GC the goroutines. That's the purpose of this PR.